orchagent/portsorch.cpp

```diff
@@ -685,23 +694,50 @@ void PortsOrch::doPortTask(Consumer &consumer)
 
                 for (auto it = m_lanesAliasSpeedMap.begin(); it != m_lanesAliasSpeedMap.end();)
                 {
+                    bool port_created = false;
+
                     if (m_portListLaneMap.find(it->first) == m_portListLaneMap.end())
                     {
-                        if (!addPort(it->first, get<1>(it->second)))
+                        // work around to avoid syncd termination on SAI error due missing create_port SAI API
+                        // can be removed when SAI redis return NotImplemented error
+                        char *platform = getenv("platform");
+                        if (platform && strstr(platform, MLNX_PLATFORM_SUBSTRING))
                         {
-                            throw runtime_error("PortsOrch initialization failure.");
+                            if (!addPort(it->first, get<1>(it->second)))
```

Comment:
addPort() should be called for all platforms as before. Or I missed something here?

Reply:
Never mind, I mistook it with addHostIntfs().